tests/serde/mod.rs

```diff
@@ -770,36 +772,42 @@ fn utc_offset_partial() {
         &[Token::Tuple { len: 3 }, Token::TupleEnd],
         "expected offset hours",
     );
-    assert_de_tokens_error::<Compact<UtcOffset>>(
+    assert_de_tokens_error::<Readable<UtcOffset>>(
+        &[Token::Tuple { len: 3 }, Token::TupleEnd],
+        "expected offset hours",
+    );
+
+    let value = UtcOffset::from_hms(23, 0, 0).unwrap();
+    assert_de_tokens::<Compact<UtcOffset>>(
+        &value.compact(),
         &[Token::Tuple { len: 3 }, Token::I8(23), Token::TupleEnd],
-        "expected offset minutes",
     );
-    assert_de_tokens_error::<Compact<UtcOffset>>(
+    let value = UtcOffset::from_hms(23, 0, 0).unwrap();
+    assert_de_tokens::<Readable<UtcOffset>>(
+        &value.readable(),
+        &[Token::Tuple { len: 3 }, Token::I8(23), Token::TupleEnd],
+    );
+
+    let value = UtcOffset::from_hms(23, 58, 0).unwrap();
+    assert_de_tokens::<Compact<UtcOffset>>(
+        &value.compact(),
         &[
             Token::Tuple { len: 3 },
```

Comment:
One other thing, actually. Shouldn't `len` here be 2, not 3? Why does this pass?

Reply:
In 'Visitor<UtcOffset>', the default value of 0 is used when the number of minutes and seconds fails or none

Reply:
Sure, but `len` is indicating the number of values present, which in this case is not 3.

Reply:
ok I know, it's done